Sources/DuetBot/main.swift

```diff
@@ -1,15 +1,27 @@
+import Foundation
 import Sword
 
-let bot = Sword(token: "token")
-
-bot.editStatus(to: "online", playing: "with Sword!")
-
-bot.on(.messageCreate) { data in
-  let msg = data as! Message
-
-  if msg.content == "!ping" {
-    msg.reply(with: "Pong!")
-  }
+guard let config = FileManager.default
+    .contents(atPath: FileManager.default.currentDirectoryPath.appending("/config.json"))
+    .flatMap({ try? JSONDecoder().decode(Config.self, from: $0) }) else {
+        print("""
+        You're missing a valid config.json file in the current directory.
+        It should contain JSON which can be decoded into a Config instance.
+        """)
+        #if DEBUG
+        print("""
+        When running in debug: the target contains a copy build phase which
+        copies a config.json file at $(SRCROOT)/Sources/DuetBot. Make sure
+        that the Products directory for this target (somewhere in your
+        DerivedData) contains this file.
+        """)
+        #endif
+        exit(1)
 }
 
-bot.connect()
+Bot(token: config.token, routines:
+    PingRoutine(),
+    PongRoutine(),
+    InsultRoutine(ignoring: ["Dizzz"])
```

Comment:
Should probably look into the message object more to see if there is a way we could just tag someone instead of having to type their name without tagging them for convenience

Reply:
Yeah, I'm wondering if an @ tag expands to a `Snowflake`. I've not tried that yet, though